models/yang/sonic/sonic-sflow.yang

```diff
@@ -86,8 +96,11 @@ module sonic-sflow {
                     }
 
                     leaf agent_id {
-                       sonic-ext:custom-validation ValidateSflowAgentId;
-                       type string;
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        sonic-ext:custom-validation ValidateSflowAgentId;
+                        type string;
```

Comment:
The type leafref is defined on this node, meaning that pointed-at source type is used for this node. Please remove this redundant type definition.

Reply:
Sure